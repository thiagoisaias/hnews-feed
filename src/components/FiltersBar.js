import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Container = styled.div`
  width: 100%;
  height: 50px;
  background-color: #fff;
  display: flex;
  align-items: center;
  justify-content: center;
  box-shadow: 0px 1.5px #eee;
`;

const FilterButton = styled.button`
  display: block;
  outline: none;
  cursor: pointer;
  color: #484848;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 3px;
  padding: 6px 12px;
  font-size: 14px;
  margin: 0 8px;

  &:hover {
    background-color: #eee;
  }
`;

const FiltersBar = props => {
  const { sortByDate, sortByPopularity } = props;
  return (
    <Container>
      <FilterButton onClick={sortByPopularity}> Popularity </FilterButton>
      <FilterButton onClick={sortByDate}> Date </FilterButton>
      <FilterButton > Range </FilterButton>
    </Container>
  );
};

FiltersBar.propTypes = {
  sortByDate: PropTypes.func.isRequired,
  sortByPopularity: PropTypes.func.isRequired
};

export default FiltersBar;
