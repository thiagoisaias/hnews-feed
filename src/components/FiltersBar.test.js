import React from "react";
import Enzyme, { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import FiltersBar from "./FiltersBar";

describe("FiltersBar", () => {
  it("renders properly", () => {
    const props = { sortByDate: () => {}, sortByPopularity: () => {} };
    const FiltersBarTest = shallow(<FiltersBar {...props} />);
    expect(shallowToJson(FiltersBarTest)).toMatchSnapshot();
  });
});
