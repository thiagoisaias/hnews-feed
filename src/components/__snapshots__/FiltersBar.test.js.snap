// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersBar renders properly 1`] = `
<styled.div>
  <styled.button
    onClick={[Function]}
  >
     Popularity 
  </styled.button>
  <styled.button
    onClick={[Function]}
  >
     Date 
  </styled.button>
  <styled.button>
     Range 
  </styled.button>
</styled.div>
`;
